src/fetch_kits.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2018 the Authors
+#
+# Purpose: Fetch kit data from the Web
+#
+
+import yaml
+from lib import *
+
+def global_init():
+  # environment variable required
+  try:
+    sys.path.append(os.environ['REDUX_PATH'])
+    REDUX_CONF = os.path.join(os.environ['REDUX_PATH'], 'config.yaml')
+  except:
+    trace(0,"Missing environment variable REDUX_PATH. Aborting.")
+    sys.exit()
+
+  # parse the remainder of the configuration settings
+  config = yaml.load(open(REDUX_CONF))
+
+
+def main():
+  global_init()
+  trace(0, 'Fetching kit info')
+  # Get the kits info into json.out no matter what the config says.
```

Comment:
I hadn't really intended json.out to be permanent. I used it because I was testing things a lot and didn't want to pull data from the web every time I ran a test. However, I think it's an idea worth keeping. It probably needs to go into the data directory rather than cluttering the source directory. It might be better to just improve the logic in the main program rather than building a new program to do just that? Or is the intent of this program to grow into more, such as loading kits?

Reply:
I rather like to separate the steps that require network connectivity and depend on external resources from those that can be executed locally.
The location (and name) of the json.out file should be in a single place in the lib, if we have that, it is easy to move it to the data dir.
Should we define that the data dir is always $REDUX_PATH/../data?
(I don't like any of the ways to do libdirs in python, unfortunately. It's simplest to keep libs and bins the same place, but it's not elegant.)